### ISSUE TITLE
LG-255: Provide ability to omit specified fields from JSON logging output

### ISSUE THREAD

**Issue by donbourne** [OpenLiberty/open-liberty#9862]
WAD: https://ibm.ent.box.com/file/566918779535

Need to be able to omit fields from JSON emitted by JSON logging.  eg. some users would like to not have `ibm_userDir` and `ibm_serverName` output in their JSON.

This could be done by extending our JSON field mapping capability from LG-60.
ID: #9948

Target beta:20200

**Comment by Yushan-Lin** [comment]
WAD: https://ibm.box.com/s/p21jzxtwhhjyorpg6ft4zizs055kpo1g

**Comment by ellen-lau** [comment]
Feature Test Summary: #10134

**Comment by Yushan-Lin** [comment]
Implementation: #9947

**Comment by donbourne** [comment]
> WAD: https://ibm.box.com/s/p21jzxtwhhjyorpg6ft4zizs055kpo1g

that link is out of date

**Comment by ellen-lau** [comment]
Not sure if this can be added to beta. The `isBeta` code relies on the ProductInfo class inside the `com.ibm.ws.kernel.boot` project (`com.ibm.ws.kernel.boot.core;version=latest` needs to be added to the bnd file build path of the `com.ibm.ws.logging` project), but the `com.ibm.ws.kernel.boot` bnd file already relies on `com.ibm.ws.logging` in its build path, causing a circular dependency.

**Comment by Yushan-Lin** [comment]
ID: https://github.com/OpenLiberty/open-liberty/issues/9948

**Comment by ellen-lau** [comment]
Since this is an extension of an already existing attribute, we have decided to implement this in `beta` with an environment variable `WLP_LOGGING_OMIT_JSON_FIELD_MAPPINGS` that will be. If the environment variable is set to `true`, the additional functionality is enabled.

**Comment by Yushan-Lin** [comment]
## List of Steps to complete or get approvals / sign-offs for Onboarding to the Liberty release (GM date)
 
Instructions:
 - Do the actions below and mark them complete in the checklist when they are done.
 - Make sure all feature readiness approvers put the appropriate tag on the epic to indicate their approval.
 
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
 ## TARGET COMPLETION DATE **Before Development Starts or 8 weeks before Onboarding** 
- [x] POC Design / WAD Review Scheduled (David Chang) or N/A. AHA Feature marked.
- [x] POC Design / WAD Reviewed (Feature Owner) or N/A. AHA Feature marked.
- [x] Complete any To-Do's from the POC Review, and create a new TO-DO in the AHA Feature for the "Design Approval Request" and assign it to Ian Robinson. Leave the date blank. (Feature Owner)
- [x] Design / WAD Approval (Ian Robinson) or N/A. AHA Feature marked.
- [x] No Design / No WAD Approval (Arthur De Magalhaes - cloud / Alasdair Nottingham - server) or N/A. AHA Feature marked.
- [x] AHA Feature has link to Github epic or issue. Github epic has link back to AHA Feature. (Feature Owner or Backlog Subtribe PM).
- [x] SVT Requirements identified. (Epic owner / Feature owner with SVT focal point)
- [ ] ID Requirements identified. (Epic owner / Feature owner with ID focal point)
- [x] Create a child task of the epic entitled "FAT Approval Test Summary"
 
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
 ## TARGET COMPLETION DATE **3 weeks before Onboarding** 
- [x] Identify all open source libraries that are changing or are new. Work with Legal Release Services (Cass Tucker or Release PM) to get open source cleared and approved. Or N/A. (Epic Owner).   New or changed open source impacts license and Certificate of Originality.
 
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
 ## TARGET COMPLETION DATE ** 3 weeks before Onboarding** 
- [x] All new or changed PII messages are checked into the integration branch, before the last translation shipment out. (Epic Owner)
 
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
 ## TARGET COMPLETION DATE **2 weeks before Onboarding** 
- [x] Implementation complete. (Epic owner / Feature owner) 
- [x] All function tests complete. Ready for FAT Approval. (Epic owner / Feature owner)
- [x] Review all known issues for Stop Ship. (Epic owner / Feature owner / PM)
 
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
 ## APPROVALS with TARGET COMPLETION DATE **2 to 1 week before Onboarding** 
Prereq: You must have the Design Approval or No Design Approval on the AHA Feature.
- [x] **Accessibility** - (G Scott Johnston). Accessibility testing is complete or N/A. Approver adds label focalApproved:accessibility to the Epic in Github.
- [ ] **FAT Liberty SOE** - (Kevin Smith). SOE FATS are running successfully or N/A . Approver adds label focalApproved:fat to the Epic in Github.
- [ ] **Globalization** (Marika Joannidis - Liberty / Simy Cheeran - tWAS). Translation is complete or N/A. TVT - complete or N/A. Approver adds label focalApproved:globalization to the Epic in Github.
- [ ] **ID** - (Kareen Deen). Documentation work is complete or N/A . Approver adds label focalApproved:id to the Epic in Github.
- [x] **Performance** - (Jared Anderson). Performance testing is complete with no high severity defects or N/A . Approver adds label focalApproved:performance to the Epic in Github.
- [x] **Serviceability** - (Don Bourne). Serviceability has been addressed. 
- [x] **STE** - (Swati Kasundra). STE chart deck is complete or N/A . Approver adds label focalApproved:ste to the Epic in Github.
- [x] **SVT** - (Tanya Kulik). SVT is complete or N/A . Approver adds label focalApproved:svt to the Epic in Github.
 
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
## TARGET COMPLETION DATE **1 week before Onboarding** 
- [ ] No Stop Ship issues for the feature. (Epic owner / Feature owner / Release PM)
- [ ] Ship Readiness Review and Release Notes completed (Epic owner / Feature owner / Release PM)
- [ ] Github Epic and Epic's issues are closed / complete. All PRs are committed to the master branch. (Epic owner / Feature owner / Backlog Subtribe PM)
 
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
 ## NOT REQUIRED FOR A FEATURE
- [ ] **OL Guides** - (Yee-Kang Chang). Assessment for OL Guides is complete or N/A. 
- [ ] **WDT** - (Leonard Theivendra). WDT work complete or N/A.
 
 
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
## Related Deliverables TARGET COMPLETION DATE **General Availability** 
- [ ] Blog article writeup (Epic owner / Feature owner / Laura Cowen)

**Comment by Yushan-Lin** [comment]
@gscottj This feature doesn't have UI. May I have accessibility approval?
@gecock SVT is not required for this feature. May I have the SVT approval?
@chirp1 ID content has been provided in the separate ID issue. May I have ID approval? #9948
@jhanders34 We don't expect performance hit in this feature. Please assess.
@marikaj123 There's no translation needed. May I have globalization approval?
@sabolo We have updated the feature test summary. Please review. #10134

**Comment by gscottj** [comment]
This feature does not add or change any end user interfaces. No accessibility verification testing required.

**Comment by Yushan-Lin** [comment]
@marikaj123 No translation is needed. May I have globalization approval?

**Comment by skasund** [comment]
L2 has requested STE slides for this feature.  The STE template can be found at the link below: 

https://ibm.box.com/s/1an42g7zdgmaj84w7dft0indqfgi8ffm

Please upload the completed slides to the same STE Archive BOX folder.  Thanks!

**Comment by Yushan-Lin** [comment]
@skasund I have uploaded the STE slides here: https://ibm.ent.box.com/file/601522903193

**Comment by skasund** [comment]
@Yushan-Lin  Thanks for the slides.  I've approved the feature.

**Comment by ellen-lau** [comment]
@jhanders34 There are no significant change in performance for this feature. May I get your approval on performance? Thanks!

**Comment by ellen-lau** [comment]
@chirp1 ID content has been provided in the separate ID issue #9948. May I have ID approval? Thanks!

**Comment by Yushan-Lin** [comment]
**1. WAD -- does the WAD identify the most likely problems customers will see and identify how the feature will enable them to diagnose and solve those problems without resorting to raising a PMR? Have these issues been addressed in the implementation?**
Yes, the WAD addresses problems customers might encounter. Warning messages were implemented and provides information about issues customers might encounter.

**2. Test and Demo -- As part of the serviceability process we're asking feature teams to test and analyze common problem paths for serviceability and demo those problem paths to someone not involved in the development of the feature (eg. L2, test team, or another development team).**
a) What problem paths were tested and demonstrated?
b) Who did you demo to?
c) Do the people you demo'd to agree that the serviceability of the demonstrated problem scenarios is sufficient to avoid PMRs for any problems customers are likely to encounter, or that L2 should be able to quickly address those problems without need to engage L3?
**TBD**
**3. SVT -- SVT team is often the first team to try new features and often encounters problems setting up and using them. Note that we're not expecting SVT to do full serviceability testing -- just to sign-off on the serviceability of the problem paths they encountered.**
a) Who conducted SVT tests for this feature?
b) Do they agree that the serviceability of the problems they encountered is sufficient to avoid PMRs, or that L2 should be able to quickly address those problems without need to engage L3?
**TBD**
**4. Which L2 / L3 queues will handle PMRs for this feature? Ensure they are present in the contact reference file and in the queue contact summary, and that the respective L2/L3 teams know they are supporting it. Ask Don Bourne if you need links or more info.**
Salesforce: WAS L3: Liberty Log Analytics
Retain: LL3ANA, 103

**Comment by ellen-lau** [comment]
GA blog here: https://github.com/OpenLiberty/open-liberty/issues/10678

**Comment by Yushan-Lin** [comment]
@donbourne Questions for serviceability has been answered above. May I have serviceability approval?

**Comment by chirp1** [comment]
The ID issue of #9948 contains the info to be documented. Approving this epic.
